search/lib/api/wfs.js

```diff
@@ -76,6 +83,65 @@ async function getCollections (request, response) {
   }
 }
 
+/**
+ * Fetch a list of cloud collections from CMR for a provider.
+ */
+async function getCloudCollections (request, response) {
```

Comment:
There seems to be an awful lot of duplication between the new Cloud functions and the normal counterparts, which will cause it to be more difficult to maintain and update in the future.
What are the needed changes here, does it come down to just the tag key?  How could this be made more generic?

Reply:
Tag key, URL links, and sometimes calling different functions. Some are easier to be combined with the original than others.